portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -62,31 +62,13 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get BlogMessages from Datastore.
     List<BlogMessage> blogMessages = LoadAllBlogsOrLast(/*all=*/ true);
- 
+    
     // TODO: Get these from client.
     int numberOfCommentsToDisplay = 0;
     List<String> tagsToSearch = new ArrayList<String>();
  
-    // TODO: Use helper functions for lines 73-101.
- 
-    // Separate posts from replies from |blogMessages|.
-    // Add replies to messageReplies for the respective posts.
-    List<BlogMessage> blogMessagesReplies = new ArrayList<BlogMessage>();
-    for (BlogMessage message : blogMessages) {
-      if (message.getParentID() != 0) {
-        blogMessagesReplies.add(message);
-        blogMessages.remove(message);
-      }
-    }
- 
-    for (BlogMessage post : blogMessages) {
-      for (BlogMessage reply : blogMessagesReplies) {
-        if (reply.getParentID() == post.getTimestamp()) {
-          post.addReply(reply);
-        }
-      } 
-    }
- 
+    blogMessages = putRepliesWithPosts(blogMessages);
```

Comment:
I would combine line 64 and 70 into

List<BlogMessage> blogMessages = putRepliesWithPosts(LoadAllBlogsOrLast(/*all=*/true));

instead of overwriting the same variable.